core/trino-main/src/test/java/io/trino/operator/TestDistinctLimitOperator.java

```diff
@@ -73,119 +73,115 @@ public void tearDown()
         scheduledExecutor.shutdownNow();
     }
 
-    @DataProvider
-    public Object[][] dataType()
+    @Test
+    public void testDistinctLimit()
     {
-        return new Object[][] {{VARCHAR}, {BIGINT}};
+        for (boolean hashEnabled : Arrays.asList(true, false)) {
```

Comment:
That's good, in theory. Unfortunately, there's not much more benefit than that. It would be nice if IntelliJ was smart enough to be able to run a specific case, but it's not. So, constraining the test to run just the scenario you want is significantly more cumbersome. 

With this approach and the one below, it's just a matter of commenting out a line of code. With the approach below, it's even easier, as you can just copy the offending call to a new test method to play with. This is similar to how tests are structure in TestUnwrapCastInComparison or any of the datetime type/function tests (e.g., io.trino.operator.scalar.timestamp.TestTimestamp)

Reply:
> That's good, in theory. Unfortunately, there's not much more benefit than that. It would be nice if IntelliJ was smart enough to be able to run a specific case, but it's not.

Actually it is.

In IntelliJ, when i have a series of failed tests (single test method with a testng `@DataProvider`) and i click re-run on indivudual failure, the IntelliJ somehow reruns the specific configuration.

Reply:
Moreover, data providers allow test framework to produce somewhat useful exception messages.

When a test method just loops over values, or calls them sequentially, we're sometimes left with checking line numbers from the stacktrace to know where the failure comes from.
I am not saying it's hard, i can do this, but not as convenient as having the test framework tell me what was the parameters that the failure occured for

Reply:
If you have many parameters for a test then you could see from the report for what parameters tests is passing and for what not. Now you need to rerun it and you will be able to see only the first failure.

Reply:
@kokosing i think there generally is agreement on this, except few people insisting on not using data providers.
i think we will simply improve test reports readability later and if needed